src/library/dlgtrackinfo.cpp

```diff
@@ -445,19 +468,58 @@ void DlgTrackInfo::clear() {
 }
 
 void DlgTrackInfo::slotBpmDouble() {
-    spinBpm->setValue(spinBpm->value() * 2.0);
+    m_pBeatsClone->scale(Beats::DOUBLE);
+    // read back the actual value
+    double newValue = m_pBeatsClone->getBpm();
+    spinBpm->setValue(newValue);
 }
 
 void DlgTrackInfo::slotBpmHalve() {
-    spinBpm->setValue(spinBpm->value() / 2.0);
+    m_pBeatsClone->scale(Beats::HALVE);
+    // read back the actual value
+    double newValue = m_pBeatsClone->getBpm();
+    spinBpm->setValue(newValue);
 }
 
 void DlgTrackInfo::slotBpmTwoThirds() {
-    spinBpm->setValue(spinBpm->value() * (2./3.));
+    m_pBeatsClone->scale(Beats::Beats::TWOTHIRDS);
+    // read back the actual value
+    double newValue = m_pBeatsClone->getBpm();
+    spinBpm->setValue(newValue);
 }
 
 void DlgTrackInfo::slotBpmThreeFourth() {
-    spinBpm->setValue(spinBpm->value() * (3./4.));
+    m_pBeatsClone->scale(Beats::Beats::THREEFOURTHS);
+    // read back the actual value
+    double newValue = m_pBeatsClone->getBpm();
+    spinBpm->setValue(newValue);
+}
+
+void DlgTrackInfo::slotBpmClear() {
+    spinBpm->setValue(0);
+    m_pBeatsClone.clear();
+
+    bpmConst->setChecked(true);
+    bpmConst->setEnabled(m_trackHasBeatMap);
+    spinBpm->setEnabled(true);
+    bpmTap->setEnabled(true);
+}
+
+void DlgTrackInfo::slotBpmConstChanged(int state) {
+    if (state != Qt::Unchecked) {
+        // const beatgrid requested
+        if (spinBpm->value() > 0) {
+            m_pBeatsClone = BeatFactory::makeBeatGrid(m_pLoadedTrack.data(),
```

Comment:
> Do you mean the beat detector unreliably detects the first beat?

That sounds too hard. If a track starts with a "real" beat it is detected nicely. But thats an exceptional case for my collection. Many tracks starts with unusable garbage or a click that I treated as a beat. 


![beatmap1](https://cloud.githubusercontent.com/assets/1777442/12879728/7e824256-ce38-11e5-85d5-88fcf9d1c47b.png)
This is a typical beat map that needs to be tweaked. The "fist beat" is at the cue point slightly off beat 3 of the beat map. Beat 1 feels random, Beat 2 is at the place where "Some sound" starts. 
If a user shifts the beat map, he will most likely align beat 3. 

![beatmap2](https://cloud.githubusercontent.com/assets/1777442/12879791/373bbcb4-ce39-11e5-8a44-87edcf60e1db.png)
This is IMHO a well detected beat map, but with double BPM. 

![beatmap3](https://cloud.githubusercontent.com/assets/1777442/12879835/aa3a0d7e-ce39-11e5-9140-d2b3e0bf098b.png)
This is an example for unusable garbage at the start. 

Since predicting "The first beat in this" case is a lottery using th cue point may work better and results into a predictable behavior. We can improve the situation a lot if we introduce a "first beat marker" or a "tweaked beat marker". But that is an other PR.